### PR TITLE
Update PKIDeployer.create_cert()

### PR DESCRIPTION
**Pull request by edewata**
Previously during installation `PKIDeployer.create_cert()` would always call the server to generate a cert ID (using either legacy or random generator), then call `pki ca-cert-create` CLI to create a cert with that ID.

The code has been modified such that in case random ID generator is used it will no longer call the server but instead it will let the CLI generate the random cert ID locally. For legacy generator it will still call the server since legacy generator can only work inside the server.

### CODE REVIEW

File: base/server/python/pki/server/deployment/__init__.py [hunk 1]
```diff
@@ -850,10 +850,13 @@ def import_cert_request(self, subsystem, tag, request):
         request_id_generator = subsystem.config.get('dbs.request.id.generator', 'legacy')
 
         if request_id_generator == 'legacy':
-            # get a new request ID from the server
+            # call the server to generate legacy request ID
             logger.info('Creating request ID for %s cert', tag)
             request.systemCert.requestID = self.client.createRequestID(request)
             logger.info('- request ID: %s', request.systemCert.requestID)
+        else:
```
Comment (ckelleyRH) [inline base/server/python/pki/server/deployment/__init__.py:857]: Do we need `elif request_id_generator == 'random'`?

File: base/server/python/pki/server/deployment/__init__.py [hunk 2]
```diff
@@ -1351,10 +1353,16 @@ def import_perm_sslserver_cert(self, instance, cert):
 
     def create_cert(self, subsystem, tag, request):
 
-        # get a new cert ID from the server
-        logger.info('Creating cert ID for %s cert', tag)
-        request.systemCert.certID = self.client.createCertID(request)
-        logger.info('- cert ID: %s', request.systemCert.certID)
+        cert_id_generator = subsystem.config.get('dbs.cert.id.generator', 'legacy')
+
+        if cert_id_generator == 'legacy':
```
Comment (ckelleyRH) [inline base/server/python/pki/server/deployment/__init__.py:1358]: As above.

**Review by ckelleyRH — COMMENTED** [review]
Just one question about what happens if we are not using the legacy generator in 2 places.

File: base/server/python/pki/server/deployment/__init__.py [hunk 3]
```diff
@@ -866,7 +869,7 @@ def import_cert_request(self, subsystem, tag, request):
             adjust_validity=request.systemCert.adjustValidity)
 
         if request_id_generator == 'random':
```
Comment (fmarco76) [inline base/server/python/pki/server/deployment/__init__.py:871]: I think this should `!= legacy`. If in the future we will add additional methods then they should be managed like the _random_ case in the java code and here you get the value in any case

File: base/server/python/pki/server/deployment/__init__.py (hunk 1 above)
Comment (fmarco76) [reply base/server/python/pki/server/deployment/__init__.py:857]: I think we have to distinguish between

- legacy -> managed by the server
- all the other -> managed without the server

Even for the moment "all the other" contains only the `random` case I would not limit here so you can implement others without change the python code

**Review by fmarco76 — APPROVED** [review]
LGTM

File: base/server/python/pki/server/deployment/__init__.py (hunk 1 above)
Comment (edewata) [reply base/server/python/pki/server/deployment/__init__.py:857]: Yeah, I don't think in the future we will create a new ID generator that can only run inside the server again, so this should cover all future cases (I updated the comment).

File: base/server/python/pki/server/deployment/__init__.py (hunk 2 above)
Comment (edewata) [reply base/server/python/pki/server/deployment/__init__.py:1358]: Similarly, the `else` should cover all future cases. I updated the comment to make it more general.

File: base/server/python/pki/server/deployment/__init__.py (hunk 3 above)
Comment (edewata) [reply base/server/python/pki/server/deployment/__init__.py]: Good point! I updated the code.

**Review by fmarco76 — APPROVED** [review]
LGTM

**Review by ckelleyRH — APPROVED** [review]
LGTM

**Comment by edewata** [comment]
@ckelleyRH @fmarco76 Thanks!

**Pull request merged**